web/Auth/ClaimsTransformer.cs

```diff
@@ -32,14 +32,20 @@ public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
             throw new Exception("Authentication failed, non-Google identity type");
 
         var googleAccountId = existingClaimsIdentity.Claims.First(c => c.Type == ClaimTypes.NameIdentifier).Value;
+        var googlePhotoUrl = existingClaimsIdentity.Claims.First(c => c.Type == "urn:google:picture").Value;
 
         // Find the user in the DB
         var databaseUser = await _context.Users.FirstOrDefaultAsync(u => u.GoogleAccountId == googleAccountId);
         if (databaseUser == null)
         {
             // If they don't exist yet, create them
-            databaseUser = await CreateUserInDatabase(googleAccountId, existingClaimsIdentity);
-            _logger.LogInformation("Created new user '{UserEmail}' in database with id {UserId}", databaseUser.GoogleEmail, databaseUser.Id);
+            databaseUser = await CreateUserInDatabase(googleAccountId, googlePhotoUrl, existingClaimsIdentity);
+            _logger.LogInformation("Created new user '{UserEmail}' in database with id {UserId}",
+                databaseUser.GoogleEmail, databaseUser.Id);
+        }
+        else
+        {
+            await UpdateUserProfilePhoto(databaseUser, googlePhotoUrl);

```

Comment:
Is this in order to update existing users?

Reply:
Yes!

Reply:
could you make the else be an else if instead then, checking that if that database user already has that googlePhotoUrl?

Reply:
Ah yeah good idea!